facebook/delphiFacebook/R/dates.R

```diff
@@ -0,0 +1,164 @@
+#' 
+#' @return Date
+#' 
+#' @importFrom lubridate floor_date
+#' 
+#' @export
+start_of_prev_full_month <- function(date) {
+  return(floor_date(date, "month") - months(1))
+}
+
+#' Get the date of the last day of the previous month
+#'
+#' @param date Date of interest
+#' 
+#' @return Date
+#' 
+#' @importFrom lubridate ceiling_date days
+#' 
+#' @export
+end_of_prev_full_month <- function(date) {
+  first_day_of_next_month <- ceiling_date(date, "month")
+  
+  if (first_day_of_next_month != date) {
+    first_day_of_next_month <- floor_date(date, "month")
+  }
+  
+  return(first_day_of_next_month - days(1))
+}
+
+#' Get the date range specifying the previous month
+#'
+#' @param date Date of interest
+#' 
+#' @return list of two Dates
+#' 
+#' @export
+get_range_prev_full_month <- function(date = Sys.Date()) {
+  eom <- end_of_prev_full_month(date)
+  
+  if (eom == date) {
+    som <- start_of_prev_full_month(date + months(1))
+  } else {
+    som <- start_of_prev_full_month(date)
+  }
+  
+  return(list(som, eom))
+}
+
+#' Get the date of the first day of the previous epiweek
+#'
+#' @param date Date of interest
+#' 
+#' @return Date
+#' 
+#' @importFrom lubridate floor_date weeks
+#' 
+#' @export
+start_of_prev_full_week <- function(date) {
+  return(floor_epiweek(date) - weeks(1))
+}
+
+#' Get the date of the last day of the previous epiweek
+#'
+#' @param date Date of interest
+#' 
+#' @return Date
+#' 
+#' @importFrom lubridate days
+#' 
+#' @export
+end_of_prev_full_week <- function(date) {
+  first_day_of_next_week <- ceiling_epiweek(date)
+  
+  if (first_day_of_next_week != date) {
+    first_day_of_next_week <- floor_epiweek(date)
+  }
+  
+  return(first_day_of_next_week - days(1))
+}
+
+#' Get the date range specifying the previous week
+#'
+#' @param date Date of interest
+#' 
+#' @return list of two Dates
+#' 
+#' @importFrom lubridate weeks
+#' 
+#' @export
+get_range_prev_full_week <- function(date = Sys.Date()) {
+  eow <- end_of_prev_full_week(date)
+  
+  if (eow == date) {
+    sow <- start_of_prev_full_week(date + weeks(1))
+  } else {
+    sow <- start_of_prev_full_week(date)
+  }
+  
+  return(list(sow, eow))
+}
+
+#' Get the date range specifying the previous full time period
+#'
+#' @param date Date of interest
+#' @param weekly_or_monthly_flag string "week" or "month" indicating desired
+#' time period to aggregate over
+#' 
+#' @return list of two Dates
+#' 
+#' @importFrom lubridate ymd_hms as_date
+#' 
+#' @export
+get_range_prev_full_period <- function(date = Sys.Date(), weekly_or_monthly_flag) {
+  if (weekly_or_monthly_flag == "month") {
+    # Get start and end of previous full month.
+    date_period_range = get_range_prev_full_month(date)
+  } else if (weekly_or_monthly_flag == "week") {
+    # Get start and end of previous full epiweek.
+    date_period_range = get_range_prev_full_week(date)
+  } else {
+    stop("invalid time period selection; must indicate either 'month' or 'week'")
+  }
```

Comment:
```suggestion
get_range_prev_full_period <- function(date = Sys.Date(), weekly_or_monthly_flag = c("month", "week")) {
  weekly_or_monthly_flag <- match.arg(weekly_or_monthly_flag)

  if (weekly_or_monthly_flag == "month") {
    # Get start and end of previous full month.
    date_period_range = get_range_prev_full_month(date)
  } else if (weekly_or_monthly_flag == "week") {
    # Get start and end of previous full epiweek.
    date_period_range = get_range_prev_full_week(date)
  }
```

`match.arg` is the idiomatic way to handle enumerated arguments like this, and automatically raises an error when the arg doesn't match the listed options

Reply:
+1